RecoLocalFastTime/FTLClusterizer/src/MTDThresholdClusterizer.cc

```diff
@@ -191,16 +196,53 @@ void MTDThresholdClusterizer::clear_buffer(RecHitIterator itr) { theBuffer.clear
 //----------------------------------------------------------------------------
 //! \brief Copy FTLRecHit into the buffer, identify seeds.
 //----------------------------------------------------------------------------
-void MTDThresholdClusterizer::copy_to_buffer(RecHitIterator itr) {
+void MTDThresholdClusterizer::copy_to_buffer(RecHitIterator itr, const MTDGeometry* geom, const MTDTopology* topo) {
+  MTDDetId mtdId = MTDDetId(itr->detid());
   int row = itr->row();
   int col = itr->column();
+  int SubDet = 0;
   float energy = itr->energy();
   float time = itr->time();
   float timeError = itr->timeError();
+  float position = itr->position().second;
+  LocalError local_error(0, 0, 0);
+  GlobalPoint global_point(0, 0, 0);
+  if (mtdId.mtdSubDetector() == MTDDetId::BTL) {
+    SubDet = 1;
+    BTLDetId id = itr->id();
+    DetId geoId = id.geographicalId((BTLDetId::CrysLayout)topo->getMTDTopologyMode());
+    const auto& det = geom->idToDet(geoId);
+    const ProxyMTDTopology& topoproxy = static_cast<const ProxyMTDTopology&>(det->topology());
+    const RectangularMTDTopology& topol = static_cast<const RectangularMTDTopology&>(topoproxy.specificTopology());
+    MeasurementPoint mp(row, col);
+    LocalPoint lp_ctr = topol.localPosition(mp);
+    float z_offset =
+        (((BTLBarLength / 2.) - position) /
+         10.);  //need to add the offset as the distance from centre to edge (z direction in global and y direction in local)
+    LocalPoint lp(lp_ctr.x(), lp_ctr.y() + z_offset, lp_ctr.z());
+    GlobalPoint gp = det->toGlobal(lp);
+    global_point = gp;
+    BTLRecHitsErrorEstimatorIM I(det, lp);
+    LocalError err_modified = I.localError();
```

Comment:
temporary seems unnecessary, just assign directly

Reply:
Hi,

Just to make sure, instead of doing 
GlobalPoint gp = det->toGlobal(lp);
global_point = gp;

I should just do:
global_point = det->toGlobal(lp);

Is that correct?

Thanks!

Reply:
yes

Reply:
here you should also do:
`local_error = I.localError();`